owncloudApp/src/androidTest/java/com/owncloud/android/shares/ui/usecases/CreatePublicShareTest.kt

```diff
@@ -194,32 +196,56 @@ class CreatePublicShareTest {
         loadCapabilitiesSuccessfully()
         loadSharesSuccessfully(arrayListOf())
 
+        // Create share
+        onView(withId(R.id.addPublicLinkButton)).perform(click())
+
+        // Force Espresso to wait until dialog fragment is visible
+        val createShareFragmentIdlingResource = registerCreateShareFragmentAsIdlingResource()
+
         val newPublicShare = publicShares[0]
+        savePublicShare(newPublicShare)
 
-        createPublicShareSuccesfully(newPublicShare, arrayListOf(newPublicShare))
+        // New share properly created
+        sharesLiveData.postValue(
+            Resource.success(
+                arrayListOf(newPublicShare)
+            )
+        )
 
         // Check whether the dialog to create the public share has been properly closed
         onView(withText(R.string.share_via_link_create_title)).check(doesNotExist())
         onView(withText(newPublicShare.name)).check(matches(isDisplayed()))
+
+        unregisterCreateShareFragmentAsIdlingResource(createShareFragmentIdlingResource)
     }
 
     @Test
     fun createPublicShareWithAlreadyExistingShares() {
         loadCapabilitiesSuccessfully()
-
         val existingPublicShares = publicShares.take(2) as ArrayList<OCShare>
-
         loadSharesSuccessfully(
             existingPublicShares
         )
 
+        onView(withId(R.id.addPublicLinkButton)).perform(click())
+
+        val createShareFragmentIdlingResource = registerCreateShareFragmentAsIdlingResource()
+
         val newPublicShare = publicShares[2]
+        savePublicShare(newPublicShare)
 
-        createPublicShareSuccesfully(newPublicShare, publicShares)
+        // New share properly created
+        sharesLiveData.postValue(
+            Resource.success(
+                publicShares
+            )
+        )
 
         // Check whether the dialog to create the public share has been properly closed
         onView(withText(R.string.share_via_link_create_title)).check(doesNotExist())
         onView(withText(newPublicShare.name)).check(matches(isDisplayed()))
+
+        unregisterCreateShareFragmentAsIdlingResource(createShareFragmentIdlingResource)
```

Comment:
why don't you include this statement in a "@After" method??

Reply:
Because the idlingresource to unregister is created for each test and not shared between different ones.

Reply:
"@before" and "@after" are functions are called before and after running one single test.

My idea would be to have your tests as following:

```
private lateinit var idlingResource: <Your type>

@Before
fun setUp() {
    idlingResource = // call constructor
    //register ...
}

@After  
fun tearDown() {
    // unregister
}
```

Reply:
Yes, I know what you mean, I tried that way but the tests were not working fine. Let me recheck it...

Reply:
> @Before
fun setUp() {
    idlingResource = // call constructor
    //register ...
}

I'm not using this because I register the idling resources in the middle of the tests, after executing other lines of code.

If you see the [Expresso idling resources docs](https://developer.android.com/training/testing/espresso/idling-resource#create-own) , there's two different ways to implement it (see below), I decided to use the second one and that's why I do not use `@After` method either.

<img width="704" alt="Screenshot 2019-06-11 at 14 27 11" src="https://user-images.githubusercontent.com/15261363/59271920-1c838180-8c55-11e9-92e5-42914de7015a.png">